src/main/java/ch/njol/skript/effects/EffBreakNaturally.java

```diff
@@ -1,66 +1,94 @@
-import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.SyntaxStringBuilder;
 import ch.njol.util.Kleenean;
+import org.bukkit.block.Block;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.log.runtime.SyntaxRuntimeErrorProducer;
 
 @Name("Break Block")
-@Description({"Breaks the block and spawns items as if a player had mined it",
-		"\nYou can add a tool, which will spawn items based on how that tool would break the block ",
-		"(ie: When using a hand to break stone, it drops nothing, whereas with a pickaxe it drops cobblestone)"})
-@Examples({"on right click:", "\tbreak clicked block naturally",
-		"loop blocks in radius 10 around player:", "\tbreak loop-block using player's tool",
-		"loop blocks in radius 10 around player:", "\tbreak loop-block naturally using diamond pickaxe"})
+@Description({
+	"Breaks the block and spawns items as if a player had mined it",
+	"You can add a tool, which will spawn items based on how that tool would break the block " +
+	"(i.e. using a hand to break stone drops nothing, whereas using a pickaxe drops cobblestone)"})
+@Examples({
+	"on right click:",
+		"\tbreak clicked block naturally",
+	"loop blocks in radius 10 around player:",
+		"\tbreak loop-block using player's tool",
+	"loop blocks in radius 10 around player:",
+		"\tbreak loop-block naturally using diamond pickaxe"
+})
 @Since("2.4")
-public class EffBreakNaturally extends Effect {
+public class EffBreakNaturally extends Effect implements SyntaxRuntimeErrorProducer {
 	
 	static {
 		Skript.registerEffect(EffBreakNaturally.class, "break %blocks% [naturally] [using %-itemtype%]");
 	}
-	
-	@SuppressWarnings("null")
+
+	private Node node;
 	private Expression<Block> blocks;
-	@Nullable
-	private Expression<ItemType> tool;
-	
-	@SuppressWarnings({"unchecked", "null"})
+	private @Nullable Expression<ItemType> tool;
+
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final SkriptParser.ParseResult parser) {
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parser) {
+		node = getParser().getNode();
 		blocks = (Expression<Block>) exprs[0];
 		tool = (Expression<ItemType>) exprs[1];
 		return true;
 	}
 	
 	@Override
-	protected void execute(final Event e) {
-		ItemType tool = this.tool != null ? this.tool.getSingle(e) : null;
-		for (Block block : this.blocks.getArray(e)) {
-			if (tool != null) {
-				ItemStack is = tool.getRandom();
-				if (is != null)
-					block.breakNaturally(is);
-				else
+	protected void execute(Event event) {
+		ItemType itemType = null;
+		if (this.tool != null) {
+			ItemType tool = this.tool.getSingle(event);
+			if (tool == null) {
+				warning("The provided tool was not set, so defaulted to nothing.");
+			} else {
```

Comment:
this one doesn't need to exist imo
if nothing is provided, and nothing is used, then that's business as usual.